src/vellum/workflows/emitters/vellum_emitter.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import time
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
+
+import httpx
+
+from vellum.workflows.emitters.base import BaseWorkflowEmitter
+from vellum.workflows.events.types import default_serializer
+from vellum.workflows.events.workflow import WorkflowEvent
+from vellum.workflows.state.base import BaseState
+
+# To protect against circular imports
+if TYPE_CHECKING:
+    from vellum.workflows.state.context import WorkflowContext
+
+logger = logging.getLogger(__name__)
+
+
+class VellumEmitter(BaseWorkflowEmitter):
+    """
+    Emitter that sends workflow events to Vellum's infrastructure for monitoring
+    externally hosted SDK-powered workflows.
+
+    Usage:
+        class MyWorkflow(BaseWorkflow):
+            emitters = [VellumEmitter]
+
+    The emitter will automatically use the same Vellum client configuration
+    as the workflow it's attached to.
+    """
+
+    def __init__(
+        self,
+        *,
+        timeout: Optional[float] = 30.0,
+        max_retries: int = 3,
+    ):
+        """
+        Initialize the VellumEmitter.
+
+        Args:
+            timeout: Request timeout in seconds.
+            max_retries: Maximum number of retry attempts for failed requests.
+        """
+        self._context: Optional["WorkflowContext"] = None
+        self._timeout = timeout
+        self._max_retries = max_retries
+        self._events_endpoint = "events"  # TODO: make this configurable with the correct url
+
+    # For now the user will need to register workflow context with the emitter via dependency injection
+    def register_context(self, context: "WorkflowContext") -> None:
```

Comment:
[nit] let's actually put this on `BaseWorkflowEmitter`, could see this being useful for the others too. let's also set the `self._context: Optional["WorkflowContext"] = None` on the base class